assets/src/apps/authoring/store/parts/actions/updatePart.ts

```diff
@@ -45,15 +45,19 @@ export const updatePart = createAsyncThunk(
       const sequence = selectSequence(rootState);
       const sequenceEntry = findInSequenceByResourceId(sequence, activityClone.id);
       const activitySequenceId = sequenceEntry?.custom.sequenceId;
-      if (authorPart && partDef.type === 'janus-hub-spoke') {
+      if (
+        authorPart &&
+        partDef.type === 'janus-hub-spoke' &&
+        payload?.changes?.custom?.spokeItems
+      ) {
         try {
           //for hub & spoke, we need to create flowchart path automatically based on the spoke destinations
           let paths =
             activityClone.authoring.flowchart?.paths?.filter(
               (path: any) => path.type === 'correct',
             ) || [];
           const flowchartPaths =
-            partDef?.custom?.spokeItems?.map((spoke: any) => {
+            payload?.changes?.custom?.spokeItems?.map((spoke: any) => {
```

Comment:
when a new spoke is added and it's the `payload` variable that contains the latest changes.